cassandraprotocol/frame/decoder.go

```diff
@@ -19,26 +19,22 @@ func (c *Codec) Decode(source io.Reader) (*Frame, error) {
 	} else {
 		header := &Header{
 			Version:          rawHeader.Version,
-			StreamId:         int16(rawHeader.StreamId & 0xFFFF),
+			StreamId:         rawHeader.StreamId,
 			TracingRequested: body.TracingId != nil || rawHeader.Flags&cassandraprotocol.HeaderFlagTracing != 0,
 		}
 		return &Frame{Header: header, Body: body}, nil
 	}
 }
 
-// A low-level representation of a frame header, as it is parsed from an encoded frame.
-type RawHeader struct {
-	IsResponse bool
-	Version    cassandraprotocol.ProtocolVersion
-	Flags      cassandraprotocol.HeaderFlag
-	StreamId   uint16
-	OpCode     cassandraprotocol.OpCode
-	BodyLength int32
-}
-
-func (r RawHeader) String() string {
-	return fmt.Sprintf("{response: %v, version: %v, flags: %08b, stream id: %v, opcode: %v, body length: %v}",
-		r.IsResponse, r.Version, r.Flags, r.StreamId, r.OpCode, r.BodyLength)
+// DecodeRaw decodes the header and reads the body as raw bytes, returning a RawFrame.
+func (c *Codec) DecodeRaw(source io.Reader) (*RawFrame, error) {
```

Comment:
I think we need to harmonize method names a bit, it's confusing from an API perspective. However this can be done later in a follow-up PR. My suggestion:

<table>
<tr>
<th>Old name</th><th>New name</th>
</tr>
<tr><td colspan="2"><b>Methods dealing with Frame<b></td></tr>
<tr><td>Encode</td><td>EncodeFrame</td></tr>
<tr><td>Decode</td><td>DecodeFrame</td></tr>
<tr><td colspan="2"><b>Methods dealing with RawFrame<b></td></tr>
<tr><td>EncodeRaw</td><td>EncodeRawFame</td></tr>
<tr><td>DecodeRaw</td><td>DecodeRawFrame</td></tr>
<tr><td colspan="2"><b>Methods converting frames<b></td></tr>
<tr><td>ConvertToRawFrame</td><td>ConvertToRawFrame</td></tr>
<tr><td> N/A </td><td>ConvertToFrame</td></tr>
<tr><td colspan="2"><b>Methods dealing with headers<b></td></tr>
<tr><td> N/A </td><td>EncodeHeader</td></tr>
<tr><td> N/A </td><td>EncodeRawHeader</td></tr>
<tr><td> N/A </td><td>DecodeHeader</td></tr>
<tr><td>DecodeHeader</td><td>DecodeRawHeader</td></tr>
<tr><td colspan="2"><b>Methods dealing with bodies<b></td></tr>
<tr><td> N/A (some private methods available) </td><td>EncodeBody (with and without compression)</td></tr>
<tr><td> N/A (some private methods available) </td><td>DecodeBody (with and without compression)</td></tr>
<tr><td> N/A </td><td>EncodeRawBody</td></tr>
<tr><td>ReadBody</td><td>DecodeRawBody (not sure we need DiscardBody)</td></tr>
</table>

Reply:
I agree, I'll change the names of the existing methods and we can create the missing ones later.

Reply:
done